src/Core/Framework/Adapter/Cache/Http/CacheHashService.php

```diff
@@ -139,6 +139,15 @@ private function isCacheHashRequired(Request $request, SalesChannelContext $sale
             return true;
         }
 
+        // check if cache relevant cookies are set
+        foreach ($this->cookies as $cookie) {
+            if (!$request->cookies->has($cookie)) {
+                continue;
+            }
+
+            return true;
```

Comment:
```suggestion
            if ($request->cookies->has($cookie)) {
                return true;
            } 
```

I guess it is only necessary to know if one of the cookies is part of the request?